File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    var answer = [];
    for(let i=0; i<commands.length; i++){
        let arr = array.slice(commands[i][0]-1,commands[i][1]);
        arr.sort((a,b)=> a-b);
        if(arr.length >= commands[i][2]){
            answer.push(arr[commands[i][2]-1]);
        }
        else{
            answer.push(arr[0]);
        }
    }
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
            for(int i=0; i<commands.length; i++){
                int [] sorted = Arrays.copyOfRange(array, commands[i][0]-1,commands[i][1]);
                Arrays.sort(sorted);
                answer[i] = sorted[commands[i][2]-1];
                }
            return answer;
    }
}